### PR TITLE
Split float/double instantiation for k-select; updated perf tests to run on native fp type

### PR DESCRIPTION
**Pull request by amgrigoriev**

### CODE REVIEW

**Review by KulikovNikita — APPROVED** [review]

**Review by makart19 — APPROVED** [review]

File: cpp/oneapi/dal/backend/primitives/selection/kselect_by_rows_impl.hpp [hunk 1]
```diff
@@ -73,9 +75,6 @@ kselect_by_rows<Float>::kselect_by_rows(sycl::queue& queue,
 
 #define INSTANTIATE(F) template class kselect_by_rows<F>;
```
Comment (rlnx) [inline cpp/oneapi/dal/backend/primitives/selection/kselect_by_rows_impl.hpp:76]: I suggest removing macro, and instantiate `template class kselect_by_rows<F>` directly in cpps.

File: cpp/oneapi/dal/backend/primitives/selection/kselect_by_rows_impl_dbl_dpc.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/backend/primitives/selection/kselect_by_rows_impl.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+#ifdef ONEDAL_DATA_PARALLEL
```
Comment (rlnx) [inline cpp/oneapi/dal/backend/primitives/selection/kselect_by_rows_impl_dbl_dpc.cpp:21]: Can be omitted as file already has `_dpc` sufix

File: cpp/oneapi/dal/backend/primitives/selection/kselect_by_rows_impl_flt_dpc.cpp [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/backend/primitives/selection/kselect_by_rows_impl.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+#ifdef ONEDAL_DATA_PARALLEL
```
Comment (rlnx) [inline cpp/oneapi/dal/backend/primitives/selection/kselect_by_rows_impl_flt_dpc.cpp:21]: Can be omitted as file already has `_dpc` sufix

**Review by rlnx — APPROVED** [review]

**Pull request merged**